### PR TITLE
Add a new method atom::getBond(), that allows testing as to whether t…

### PR DESCRIPTION
**Pull request by johnmay**
…wo atoms are connected outside of the AtomContainer.

### CODE REVIEW

File: base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java [hunk 1]
```diff
@@ -204,8 +204,21 @@
...
 
+    /**
+     * Get the number of explicit bonds connected to this atom.
+     * @return the total bond count
+     */
     int getBondCount();
 
+    /**
+     * Returns the bond connecting 'this' atom to the provided atom. If the
+     * atoms are not bonded, null is return
```
Comment (egonw) [inline base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java]: Missing period at the end?

File: base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java [hunk 2]
```diff
@@ -204,8 +204,21 @@
...
+     * @param atom the other atom
+     * @return the bond connecting the atoms
+     * @throws UnsupportedOperationException thrown if the bonds are not known
+     */
+    IBond getBond(IAtom atom);
```
Comment (egonw) [inline base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java]: So, the CDK design has the idea that an electron contains has one or more atoms, if two or more, it's a bond... there is indeed a lot of code that assumes two-atom bonds, but this method really formalizes that... This is not a arguing against it, and I'm just pondering where we should head with the CDK...

**Comment by egonw** [comment]
@johnmay, sorry, I had to revert this pull request... but not sure how to reopen it...

**Comment by johnmay** [comment]
It's fine will do another one, my bad should have checked it more locally.

File: base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java (hunk 2 above)
Comment (johnmay) [reply base/interfaces/src/main/java/org/openscience/cdk/interfaces/IAtom.java]: Technically it doesn't, but the implemented it is does as I only check the begin an end. The idea of having multiple atoms in an electron container is cute but really not very well thought or implemented. As described before the better way of doing multicenter bonds is as a layer on top of a valid but inaccurate valence bond representation ala IUPAC InChI discussions and Alex Clark's recent blog post.

**Pull request merged**